rocker/shiny:4.0.2 can't start with user set to shiny

According to [the readme for the rocker/shiny image](https://github.com/rocker-org/shiny/#warnings) it should be possible to start a contained with `--user shiny`. But when I try this with the latest release (`rocker/shiny:4.0.2`)...

```
bash> docker run --user shiny -p 3838:3838 --rm rocker/shiny:4.0.2
```

... the container doesn't start and I get the following error:

```
s6-mkdir: warning: unable to mkdir /var/run/s6: Permission denied
```

**Is it now recommended to run as root instead or is there now some other way to run as the shiny user?**

Running with `--user root` works as expected:

```
bash> docker run --user root -p 3838:3838 --rm rocker/shiny:4.0.2
```
```
[s6-init] making user provided files available at /var/run/s6/etc...exited 0.
[s6-init] ensuring user provided files have correct perms...exited 0.
[fix-attrs.d] applying ownership & permissions fixes...
[fix-attrs.d] done.
[cont-init.d] executing container initialization scripts...
[cont-init.d] done.
[services.d] starting services

*** warning - no files are being watched ***
[services.d] done.
[2020-09-10T06:59:34.982] [INFO] shiny-server - Shiny Server v1.5.14.948 (Node.js v12.18.0)
[2020-09-10T06:59:34.986] [INFO] shiny-server - Using config file "/etc/shiny-server/shiny-server.conf"
[2020-09-10T06:59:35.034] [WARN] shiny-server - Running as root unnecessarily is a security risk! You could be running more securely as non-root.
[2020-09-10T06:59:35.041] [INFO] shiny-server - Starting listener on http://[::]:3838
```

### System info ###

This was done on a 2018 Macbook pro running MacOs Mojave (10.14.6) and Docker Desktop (2.3.0.4).

Now another issue, but easily fixed. When accessing a shiny-app I get the following (running container as user `shiny`):

```
When accessing an app through the shiny-server I now get this error: 
[2021-09-22T11:27:57.528] [ERROR] shiny-server - User bookmark state directory creation failed: /var/lib/shiny-server/bookmarks/shiny
[2021-09-22T11:27:57.529] [WARN] shiny-server - ENOENT: no such file or directory, stat '/var/lib/shiny-server/bookmarks/shiny'
[2021-09-22T11:27:57.529] [INFO] shiny-server - Error getting worker: Error: The application exited during initialization.
```

Adding `RUN mkdir -p /var/lib/shiny-server/bookmarks/shiny` to the `Dockerfile` fixed it.

--- a/dockerfiles/ml_4.0.0.Dockerfile
+++ b/dockerfiles/ml_4.0.0.Dockerfile
@@ -5,7 +5,7 @@ LABEL org.opencontainers.image.licenses="GPL-2.0-or-later" \
       org.opencontainers.image.vendor="Rocker Project" \
       org.opencontainers.image.authors="Carl Boettiger <cboettig@ropensci.org>"
 
-ENV S6_VERSION=v2.0.0.1
+ENV S6_VERSION=v2.1.0.2
 ENV RSTUDIO_VERSION=1.3.959
 ENV DEFAULT_USER=rstudio
 ENV PANDOC_VERSION=default

--- a/dockerfiles/ml_4.0.1.Dockerfile
+++ b/dockerfiles/ml_4.0.1.Dockerfile
@@ -5,7 +5,7 @@ LABEL org.opencontainers.image.licenses="GPL-2.0-or-later" \
       org.opencontainers.image.vendor="Rocker Project" \
       org.opencontainers.image.authors="Carl Boettiger <cboettig@ropensci.org>"
 
-ENV S6_VERSION=v2.0.0.1
+ENV S6_VERSION=v2.1.0.2
 ENV RSTUDIO_VERSION=1.3.959
 ENV DEFAULT_USER=rstudio
 ENV PANDOC_VERSION=default

--- a/dockerfiles/ml_4.0.2.Dockerfile
+++ b/dockerfiles/ml_4.0.2.Dockerfile
@@ -5,7 +5,7 @@ LABEL org.opencontainers.image.licenses="GPL-2.0-or-later" \
       org.opencontainers.image.vendor="Rocker Project" \
       org.opencontainers.image.authors="Carl Boettiger <cboettig@ropensci.org>"
 
-ENV S6_VERSION=v2.0.0.1
+ENV S6_VERSION=v2.1.0.2
 ENV RSTUDIO_VERSION=1.3.1093
 ENV DEFAULT_USER=rstudio
 ENV PANDOC_VERSION=default

--- a/dockerfiles/rstudio-ubuntu18.04_4.0.0.Dockerfile
+++ b/dockerfiles/rstudio-ubuntu18.04_4.0.0.Dockerfile
@@ -5,7 +5,7 @@ LABEL org.opencontainers.image.licenses="GPL-2.0-or-later" \
       org.opencontainers.image.vendor="Rocker Project" \
       org.opencontainers.image.authors="Carl Boettiger <cboettig@ropensci.org>"
 
-ENV S6_VERSION=v1.21.7.0
+ENV S6_VERSION=v2.1.0.2
 ENV RSTUDIO_VERSION=1.3.959
 ENV DEFAULT_USER=rstudio
 ENV PATH=/usr/lib/rstudio-server/bin:$PATH

--- a/dockerfiles/rstudio_4.0.0.Dockerfile
+++ b/dockerfiles/rstudio_4.0.0.Dockerfile
@@ -5,7 +5,7 @@ LABEL org.opencontainers.image.licenses="GPL-2.0-or-later" \
       org.opencontainers.image.vendor="Rocker Project" \
       org.opencontainers.image.authors="Carl Boettiger <cboettig@ropensci.org>"
 
-ENV S6_VERSION=v2.0.0.1
+ENV S6_VERSION=v2.1.0.2
 ENV RSTUDIO_VERSION=1.3.959
 ENV DEFAULT_USER=rstudio
 ENV PATH=/usr/lib/rstudio-server/bin:$PATH

--- a/dockerfiles/rstudio_4.0.1.Dockerfile
+++ b/dockerfiles/rstudio_4.0.1.Dockerfile
@@ -5,7 +5,7 @@ LABEL org.opencontainers.image.licenses="GPL-2.0-or-later" \
       org.opencontainers.image.vendor="Rocker Project" \
       org.opencontainers.image.authors="Carl Boettiger <cboettig@ropensci.org>"
 
-ENV S6_VERSION=v2.0.0.1
+ENV S6_VERSION=v2.1.0.2
 ENV RSTUDIO_VERSION=1.3.959
 ENV DEFAULT_USER=rstudio
 ENV PATH=/usr/lib/rstudio-server/bin:$PATH

--- a/dockerfiles/rstudio_4.0.2.Dockerfile
+++ b/dockerfiles/rstudio_4.0.2.Dockerfile
@@ -5,7 +5,7 @@ LABEL org.opencontainers.image.licenses="GPL-2.0-or-later" \
       org.opencontainers.image.vendor="Rocker Project" \
       org.opencontainers.image.authors="Carl Boettiger <cboettig@ropensci.org>"
 
-ENV S6_VERSION=v2.0.0.1
+ENV S6_VERSION=v2.1.0.2
 ENV RSTUDIO_VERSION=1.3.1093
 ENV DEFAULT_USER=rstudio
 ENV PATH=/usr/lib/rstudio-server/bin:$PATH

--- a/dockerfiles/shiny_4.0.0.Dockerfile
+++ b/dockerfiles/shiny_4.0.0.Dockerfile
@@ -5,7 +5,7 @@ LABEL org.opencontainers.image.licenses="GPL-2.0-or-later" \
       org.opencontainers.image.vendor="Rocker Project" \
       org.opencontainers.image.authors="Carl Boettiger <cboettig@ropensci.org>"
 
-ENV S6_VERSION=v2.0.0.1
+ENV S6_VERSION=v2.1.0.2
 ENV SHINY_SERVER_VERSION=latest
 ENV PANDOC_VERSION=default
 

--- a/dockerfiles/shiny_4.0.1.Dockerfile
+++ b/dockerfiles/shiny_4.0.1.Dockerfile
@@ -5,7 +5,7 @@ LABEL org.opencontainers.image.licenses="GPL-2.0-or-later" \
       org.opencontainers.image.vendor="Rocker Project" \
       org.opencontainers.image.authors="Carl Boettiger <cboettig@ropensci.org>"
 
-ENV S6_VERSION=v2.0.0.1
+ENV S6_VERSION=v2.1.0.2
 ENV SHINY_SERVER_VERSION=latest
 ENV PANDOC_VERSION=default
 

--- a/dockerfiles/shiny_4.0.2.Dockerfile
+++ b/dockerfiles/shiny_4.0.2.Dockerfile
@@ -5,7 +5,7 @@ LABEL org.opencontainers.image.licenses="GPL-2.0-or-later" \
       org.opencontainers.image.vendor="Rocker Project" \
       org.opencontainers.image.authors="Carl Boettiger <cboettig@ropensci.org>"
 
-ENV S6_VERSION=v2.0.0.1
+ENV S6_VERSION=v2.1.0.2
 ENV SHINY_SERVER_VERSION=latest
 ENV PANDOC_VERSION=default
 

--- a/stacks/4.0.0.json
+++ b/stacks/4.0.0.json
@@ -52,7 +52,7 @@
       },
       "FROM": "rocker/r-ver:4.0.0",
       "ENV": {
-        "S6_VERSION": "v2.0.0.1",
+        "S6_VERSION": "v2.1.0.2",
         "RSTUDIO_VERSION": "1.3.959",
         "DEFAULT_USER": "rstudio",
         "PATH": "/usr/lib/rstudio-server/bin:$PATH"
@@ -115,7 +115,7 @@
       },
       "FROM": "rocker/r-ver:4.0.0",
       "ENV": {
-        "S6_VERSION": "v2.0.0.1",
+        "S6_VERSION": "v2.1.0.2",
         "SHINY_SERVER_VERSION": "latest",
         "PANDOC_VERSION": "default"
       },
@@ -204,7 +204,7 @@
       },
       "FROM": "rocker/cuda:4.0.0",
       "ENV": {
-        "S6_VERSION": "v2.0.0.1",
+        "S6_VERSION": "v2.1.0.2",
         "RSTUDIO_VERSION": "1.3.959",
         "DEFAULT_USER": "rstudio",
         "PANDOC_VERSION": "default",
@@ -275,7 +275,7 @@
       },
       "FROM": "rocker/r-ver:4.0.0-ubuntu18.04",
       "ENV": {
-        "S6_VERSION": "v1.21.7.0",
+        "S6_VERSION": "v2.1.0.2",
         "RSTUDIO_VERSION": "1.3.959",
         "DEFAULT_USER": "rstudio",
         "PATH": "/usr/lib/rstudio-server/bin:$PATH"

--- a/stacks/4.0.1.json
+++ b/stacks/4.0.1.json
@@ -38,7 +38,7 @@
       },
       "FROM": "rocker/r-ver:4.0.1",
       "ENV": {
-        "S6_VERSION": "v2.0.0.1",
+        "S6_VERSION": "v2.1.0.2",
         "RSTUDIO_VERSION": "1.3.959",
         "DEFAULT_USER": "rstudio",
         "PATH": "/usr/lib/rstudio-server/bin:$PATH"
@@ -101,7 +101,7 @@
       },
       "FROM": "rocker/r-ver:4.0.1",
       "ENV": {
-        "S6_VERSION": "v2.0.0.1",
+        "S6_VERSION": "v2.1.0.2",
         "SHINY_SERVER_VERSION": "latest",
         "PANDOC_VERSION": "default"
       },
@@ -190,7 +190,7 @@
       },
       "FROM": "rocker/cuda:4.0.1",
       "ENV": {
-        "S6_VERSION": "v2.0.0.1",
+        "S6_VERSION": "v2.1.0.2",
         "RSTUDIO_VERSION": "1.3.959",
         "DEFAULT_USER": "rstudio",
         "PANDOC_VERSION": "default",

--- a/stacks/4.0.2.json
+++ b/stacks/4.0.2.json
@@ -38,7 +38,7 @@
       },
       "FROM": "rocker/r-ver:4.0.2",
       "ENV": {
-        "S6_VERSION": "v2.0.0.1",
+        "S6_VERSION": "v2.1.0.2",
         "RSTUDIO_VERSION": "1.3.1093",
         "DEFAULT_USER": "rstudio",
         "PATH": "/usr/lib/rstudio-server/bin:$PATH"
@@ -101,7 +101,7 @@
       },
       "FROM": "rocker/r-ver:4.0.2",
       "ENV": {
-        "S6_VERSION": "v2.0.0.1",
+        "S6_VERSION": "v2.1.0.2",
         "SHINY_SERVER_VERSION": "latest",
         "PANDOC_VERSION": "default"
       },
@@ -190,7 +190,7 @@
       },
       "FROM": "rocker/cuda:4.0.2",
       "ENV": {
-        "S6_VERSION": "v2.0.0.1",
+        "S6_VERSION": "v2.1.0.2",
         "RSTUDIO_VERSION": "1.3.1093",
         "DEFAULT_USER": "rstudio",
         "PANDOC_VERSION": "default",

